src-opam/dockerfile_distro.ml

```diff
@@ -74,10 +80,20 @@ let distros = [
   `Ubuntu `V12_04; `Ubuntu `V14_04; `Ubuntu `V15_04; `Ubuntu `V15_10;
   `Ubuntu `V16_04; `Ubuntu `V16_10; `Ubuntu `V17_04; `Ubuntu `V17_10; `Ubuntu `V18_04; `Ubuntu `V18_10; `Ubuntu `V19_04; `Ubuntu `V19_10; `Ubuntu `V20_04; `Ubuntu `V20_10; `Ubuntu `V21_04;
   `Ubuntu `Latest; `Ubuntu `LTS;
-  `Cygwin `V20H2;
-  `Windows (`Mingw, `V1809); `Windows (`Mingw, `V1903); `Windows (`Mingw, `V1909); `Windows (`Mingw, `V2004); `Windows (`Mingw, `V20H2);
-  `Windows (`Msvc, `V1809); `Windows (`Msvc, `V1903); `Windows (`Msvc, `V1909); `Windows (`Msvc, `V2004); `Windows (`Msvc, `V20H2);
 ]
+let distros = List.fold_left (fun distros version ->
+                  `Cygwin version :: `Windows (`Mingw, version) :: `Windows (`Msvc, version) :: distros)
+                distros win10_releases
+
+(* https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/base-image-lifecycle *)
+let win10_release_status v ~servercore : status = match v with
+  | `V1507 | `V1511 -> `Deprecated
+  | `V1607 -> if servercore then `Active `Tier3 else `Deprecated
+  | `V1703 | `V1709 -> `Deprecated
+  | `V1809 -> `Active `Tier3
+  | `V1903 -> `Deprecated
+  | `V1909 | `V2004 | `V20H2 -> `Active `Tier3
+  | `V21H1 -> `Deprecated       (* Not yet available. *)
```

Comment:
I think better for this to be `` `Active `Tier3`` and handle the fact the container isn't yet available elsewhere (either by delaying the merge or by having a special case in docker-base-images, or special-casing it below in distro_status).

Reply:
Now we track the Microsoft support status of each version, and their availability in the Docker repository, so we correctly report that 21H1 has been released by Microsoft, but isn't available as a Docker image (yet).